scripts/configure

```diff
@@ -0,0 +1,337 @@
+#!/bin/bash -e
+
+# Copyright (c) 2023 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Usage: configure [OPTIONS] [--project=... [PROJECT OPTIONS]]
```

Comment:
should we add this to CI to not let it regress?

Reply:
Yeah, that's definitely on my TODO list, just haven't figured out what the best thing to use as a test. I guess I could just set up a lighting-app build or something like that. My longer term use case is building a sample / reference app under OpenWRT but that stuff isn't read to make public yet.

Reply:
I've added a chip-build-minimal docker image and a workflow that builds all-clusters-app on it. The workflow won't pass until the image is pushed to docker.com but here's a run against a copy of the image hosted on GitHub: https://github.com/ksperling-apple/connectedhomeip/actions/runs/4803805736/jobs/8548674304